### PR TITLE
Move management endpoints to separate HTTP port

### LINKED ISSUE #7343: Consider moving management HTTP endpoints to separate HTTP port
Quarkus since version 3 allows a separate HTTP endpoint for management endpoints (health, metrics, prometheus), see [Quarkus management interface docs](https://quarkus.io/guides/management-interface-reference).

This allows for example different authentication settings, but especially allows isolation of the management endpoints, to not expose those "to the public".

### PR DESCRIPTION
**Pull request by snazy**
Quarkus since version 3 allows a separate HTTP endpoint for management endpoints (health, metrics, prometheus), see [Quarkus management interface docs](https://quarkus.io/guides/management-interface-reference).

This allows for example different authentication settings, but especially allows isolation of the management endpoints, to not expose those "to the public".

Fixes #7343

### CODE REVIEW

File: helm/nessie/templates/deployment.yaml
Comment (adutra) [inline helm/nessie/templates/deployment.yaml:225]: FYI this will have an impact on the K8s Operator as well, I will take care of that once this PR is merged.

File: helm/nessie/templates/deployment.yaml
Comment (snazy) [reply helm/nessie/templates/deployment.yaml:232]: Added a note to CHANGELOG

**Comment by snazy** [comment]
BTW: The Quarkus issue that let both the HTTP and management server share the same port in tests will be fixed with https://github.com/quarkusio/quarkus/pull/39588 - probably worth to wait for a release with that fix.

File: helm/nessie/templates/deployment.yaml
Comment (dimas-b) [inline helm/nessie/templates/deployment.yaml:248]: Technically, a successful check on port `9000` does not guarantee that the REST endpoints are ready on port `19120`, but I guess we can live with that risk.

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java
Comment (dimas-b) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java:32]: The intention of this test is to validate that health checks work when authentication is on. Maybe rename to `TestHealthCheckWithAuthentication`?

File: servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java
Comment (dimas-b) [inline servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java:515]: Why not inject it like `@NessieClientUri`?

**Review by dimas-b — COMMENTED** [review]
LGTM overall :+1:

File: helm/nessie/templates/deployment.yaml
Comment (snazy) [reply helm/nessie/templates/deployment.yaml:248]: There is no real risk.
See `io.quarkus.vertx.http.runtime.VertxHttpRecorder#doServerStart` - the main server is started _before_ the _management_ server.

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java:32]: Yes, that was when the management endpoints were on the same port.
But the management port has no authentication configured - that's why I renamed it, because there's no authentication.

File: servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java
Comment (snazy) [reply servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java:515]: It's rather rare - so :shrug:

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java
Comment (dimas-b) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestHealthCheckNoAuthentication.java:32]: fair enough

File: servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java
Comment (dimas-b) [reply servers/quarkus-server/src/testFixtures/java/org/projectnessie/server/AbstractQuarkusEvents.java:515]: fair enough

**Review by dimas-b — APPROVED** [review]

**Pull request merged**
